app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -71,39 +70,48 @@ class VirtualMachineTemplate < ActiveRecord::Base
 
     scope :active, -> { where(state: 'active') }
 
-    scope :on_active_cs, -> do
-      joins(:tenant).
-        where(atmosphere_tenants: { active: true })
+    scope :on_active_tenant, -> do
+      joins(:tenants)
+        .where('atmosphere_tenants.active = ?', true)
     end
 
-    scope :on_cs, ->(t) { where(tenant_id: t) }
+    scope :on_tenant, ->(t) do
+      joins(:tenants)
+        .where('atmosphere_tenants.id = ?', t)
+    end
 
-    scope :on_cs_with_src, ->(t_id, source_id) do
-      joins(:tenant).
-        where(atmosphere_tenants: { tenant_id: t_id },
-              id_at_site: source_id)
+    scope :on_tenant_with_src, ->(t_id, source_id) do
+      joins(:tenants)
+        .where("atmosphere_tenants.tenant_id = '#{t_id}' AND
+        id_at_site = '#{source_id}'")
     end
 
     def uuid
-      "#{tenant.tenant_id}-tmpl-#{id_at_site}"
+      if tenants.blank?
+        "NOTENANT-tmpl-#{id_at_site}"
```

Comment:
I will keep this assignment as a "goalkeeper" solution in case the database is in an inconsistent state but I will also add the requested validation.